pkg/constraints/constraint.go

```diff
@@ -29,11 +29,14 @@ type Constraint struct {
 	// https://github.com/operator-framework/enhancements/blob/master/enhancements/compound-bundle-constraints.md
 	All *CompoundConstraint `json:"all,omitempty" yaml:"all,omitempty"`
 	Any *CompoundConstraint `json:"any,omitempty" yaml:"any,omitempty"`
-	// A note on Not: this constraint isn't particularly useful by itself.
+	// A note on Not: this constraint is not particularly useful by itself.
 	// It should be used within an All constraint alongside some other constraint type
 	// since saying "do not use any of these GVKs/packages/etc." without an alternative
-	// doesn't make sense.
+	// doesn't make sense. Both Not and None can be used for this type but None is
+	// deprecated.
 	Not *CompoundConstraint `json:"not,omitempty" yaml:"not,omitempty"`
+	// Deprecated: Use Not instead. This has been deprecated as Not has a clearer UX.
```

Comment:
What happens if both `Not` and `None` are defined simultaneously? Seems like there are a few options:
1. Ignore `None`
2. Combine `None` and `Not`
3. Error

I'd lean toward 2 or 3 I think. Whatever we choose, it would be nice to add that to the GoDoc.

Reply:
What would combining them both together look like to you?

Reply:
I vote for option 1. I think adding a comment to both fields with something like "if the values of both `None` and `Not` are defined, `Not` is preferred" would be sufficient to codify the desired API behavior (since we can't always control the marshaller being used by a client).

Reply:
Added a comment for this in the most recent push.

Reply:
Now that I see the `Parse` method again, I think it should handle pruning things down, but should also return a real error when more than one is defined so that the invoking code can bubble a warning if necessary.